### PR TITLE
Rubocop all the things

### PR DESCRIPTION
**Pull request by hunner**
Use rubocops auto-fixer and manually fix many rubocop
cops and remove unneeded ones from .rubocop.yml

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,13 +14,14 @@ group :test do
...
-  gem 'rspec-puppet', :git => 'https://github.com/rodjek/rspec-puppet.git'
+  gem 'rspec-puppet', git: 'https://github.com/rodjek/rspec-puppet.git'
   gem 'puppetlabs_spec_helper', '< 1.0'
   gem 'metadata-json-lint'
   # json_pure 2.0.2 added a requirement on ruby >= 2. We pin to json_pure 2.0.1
   # if using ruby 1.x
-  gem 'json_pure', '<=2.0.1', :require => false if RUBY_VERSION =~ /^1\./
+  gem 'json_pure', '<=2.0.1', require: false if RUBY_VERSION =~ /^1\./
   gem 'rubocop', '~> 0.41.0'
+  gem 'rubocop-rspec'
```
Comment (DavidS) [inline Gemfile]: In lieu of modulesyncing, please see https://github.com/puppetlabs/modulesync_configs/blob/master/moduleroot/Gemfile#L48-L49 for the recommended magic incantation.

**Comment by DavidS** [comment]
modulesync_configs also has a .rubocop.yml (https://github.com/puppetlabs/modulesync_configs/blob/master/moduleroot/.rubocop.yml) please take care that they stay in sync.

**Comment by hunner** [comment]
This does not pass acceptance tests :/

**Comment by gregohardy** [comment]
$ rake rubocop
Running RuboCop...
Inspecting 62 files
..............................................................

62 files inspected, no offenses detected

Hunner can we close this?

**Comment by gregohardy** [comment]
CLOUD-1474 raised to track the rubocop upgrade. This has drifted too far over time to allow it to be merged effectively.

**Pull request closed without merging**